packages/web-components/src/components/va-text-input/va-text-input.tsx

```diff
@@ -146,6 +158,16 @@ export class VaTextInput {
     }
   };
 
+  connectedCallback() {
+    i18next.on('languageChanged', () => {
+      console.log('language changed text-input');
+      forceUpdate(this.el);
+    });
+  }
```

Comment:
This is the same pattern that we're using with the `useEffect` hook in the `<Date>` component. The component forces a re-render when it detects a language change, and then the next calls to `i18next.t()` will give output using the newly set language.